zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -810,3 +686,154 @@ pub(crate) fn generate_lagrange_base_polynomial<
+
+        let t0 = a_limbs[0].clone() * b_limbs[0].clone();
+        let t1 = a_limbs[0].clone() * b_limbs[1].clone() + a_limbs[1].clone() * b_limbs[0].clone();
+        let t2 = a_limbs[0].clone() * b_limbs[2].clone()
+            + a_limbs[1].clone() * b_limbs[1].clone()
+            + a_limbs[2].clone() * b_limbs[0].clone();
+        let t3 = a_limbs[0].clone() * b_limbs[3].clone()
+            + a_limbs[1].clone() * b_limbs[2].clone()
+            + a_limbs[2].clone() * b_limbs[1].clone()
+            + a_limbs[3].clone() * b_limbs[0].clone();
+        let overflow = carry_hi_expr.clone()
+            + a_limbs[1].clone() * b_limbs[3].clone()
+            + a_limbs[2].clone() * b_limbs[2].clone()
+            + a_limbs[3].clone() * b_limbs[2].clone()
+            + a_limbs[2].clone() * b_limbs[3].clone()
+            + a_limbs[3].clone() * b_limbs[2].clone()
+            + a_limbs[3].clone() * b_limbs[3].clone();
+
+        cb.require_equal(
+            "(a * b)_lo + c_lo == d_lo + carry_lo ⋅ 2^128",
+            t0.expr() + t1.expr() * pow_of_two_expr(64) + c_lo,
+            d_lo + carry_lo_expr.clone() * pow_of_two_expr(128),
+        );
+        cb.require_equal(
+            "(a * b)_hi + c_hi + carry_lo == d_hi + carry_hi ⋅ 2^128",
+            t2.expr() + t3.expr() * pow_of_two_expr(64) + c_hi + carry_lo_expr,
+            d_hi + carry_hi_expr * pow_of_two_expr(128),
+        );
+
+        Self {
+            a,
+            b,
+            c,
+            d,
+            carry_lo,
+            carry_hi,
+            overflow,
+        }
+    }
+
+    pub(crate) fn assign(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        words: [Word; 4],
+    ) -> Result<(), Error> {
+        let (a, b, c, d) = (words[0], words[1], words[2], words[3]);
+        self.a.assign(region, offset, Some(a.to_le_bytes()))?;
+        self.b.assign(region, offset, Some(b.to_le_bytes()))?;
+        self.c.assign(region, offset, Some(c.to_le_bytes()))?;
+        self.d.assign(region, offset, Some(d.to_le_bytes()))?;
+
+        let a_limbs = split_u256_limb64(&a);
+        let b_limbs = split_u256_limb64(&b);
+        let (c_lo, c_hi) = split_u256(&c);
+        let (d_lo, d_hi) = split_u256(&d);
+
+        let t0 = a_limbs[0] * b_limbs[0];
+        let t1 = a_limbs[0] * b_limbs[1] + a_limbs[1] * b_limbs[0];
+        let t2 = a_limbs[0] * b_limbs[2] + a_limbs[1] * b_limbs[1] + a_limbs[2] * b_limbs[0];
+        let t3 = a_limbs[0] * b_limbs[3]
+            + a_limbs[1] * b_limbs[2]
+            + a_limbs[2] * b_limbs[1]
+            + a_limbs[3] * b_limbs[0];
+
+        let carry_lo = (t0 + (t1 << 64) + c_lo - d_lo) >> 128;
+        let carry_hi = (t2 + (t3 << 64) + c_hi + carry_lo - d_hi) >> 128;
+
+        self.carry_lo
+            .iter()
+            .zip(carry_lo.to_le_bytes().iter())
+            .map(|(cell, byte)| cell.assign(region, offset, Some(F::from(*byte as u64))))
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
You should be able to use `try_collect`. But use what is easier for you. Just leaving as suggestion.

Reply:
`Map` doesn't provide `try_collect`.